app/src/main/java/unipd/se18/ocrcamera/ImageProcessing.java

```diff
@@ -0,0 +1,349 @@
+        Bitmap image = Bitmap.createBitmap(matrix.width(), matrix.height(), Bitmap.Config.ARGB_8888);
+        Utils.matToBitmap(matrix, image);
+        return image;
+    }
+
+
+    /**
+     * Crop the matrix with the given rectangle
+     * @param rectangle the part of the image you want to crop
+     * @param mat the matrix you want to crop
+     * @return a matrix that contains only the rectangle
+     * @author Thomas Porro(g1), Oscar Garrido (g1)
+     */
+    public Mat crop(RotatedRect rectangle, Mat mat) {
+
+        RotatedRect rect;
+
+        //Matrices we'll use
+        Mat rotationMat = new Mat();
+        Mat rotatedImg = new Mat();
+        Mat croppedImg = new Mat();
+
+        //Get angle and size from the bounding box
+        double angle = rectangle.angle;
+        Size rect_size = rectangle.size;
+
+        //Thanks to http://felix.abecassis.me/2011/10/opencv-rotation-deskewing/
+        if (rectangle.angle < -45.) {
+            angle += 90.0;
+            double width = rect_size.width;
+            double height = rect_size.height;
+            rect_size.width = height;
+            rect_size.height = width;
+        }
+
+        //Creates the rotation matrix
+        rotationMat = getRotationMatrix2D(rectangle.center, angle, 1.0);
+
+        //Perform the affine transformation
+        Imgproc.warpAffine(mat, rotatedImg, rotationMat, mat.size(), INTER_CUBIC);
+
+        //Crop the resulting image
+        getRectSubPix(rotatedImg, rect_size, rectangle.center, croppedImg);
+        return croppedImg;
+
+    }
+
+
+    /**
+     * Converts a matrix into a Bitmap and saves it in a local directory
+     * @param matrix the matrix to be converted
+     * @param imageName the name of the file being saved
+     * @author Thomas Porro(g1), Oscar Garrido(g1)
+     */
+    private void save(Mat matrix, String imageName) {
+        final String directory = Environment.getExternalStorageDirectory() + "/" + Environment.DIRECTORY_PICTURES + "/ImageProcessingTest/" + imageName;
+        Bitmap image = conversion(matrix);
+
+        OutputStream outStream = null;
+
+        File file = new File(directory);
+        file.mkdirs();
+
+        if (file.exists()) {
+            file.delete();
+            file = new File(directory);
+        }
+        try {
+            outStream = new FileOutputStream(file);
+            image.compress(Bitmap.CompressFormat.PNG, 100, outStream);
+            outStream.flush();
+            outStream.close();
+            Log.i(TAG, "outStream closed");
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+    }
+
+
+    /**
+     * Converts the Bitmap into a grayscale matrix
+     * @param imagePath the matrix you want to convert
+     * @return the bitmap corresponding to the matrix
+     * @throws FileNotFoundException if the imagePath doesn't exist
+     * @author Oscar Garrido (g1)
+     */
+    public Mat conversion(String imagePath) throws FileNotFoundException {
```

Comment:
This method could very well be made private since it is only used for our own analysis!
The other "conversion" method is exactly the opposite of the one I wrote, I'll make sure that their name differ so that they will already describe themselves without confusion